src/Native/Runtime/windows/PalRedhawkMinWin.cpp

```diff
@@ -1267,99 +1231,440 @@ REDHAWK_PALEXPORT _Ret_maybenull_ void* REDHAWK_PALAPI PalSetWerDataBuffer(_In_
+        return 1;
+
+    pmask &= smask;
+        
+    int count = 0;
+    while (pmask)
+    {
+        if (pmask & 1)
+            count++;
+                
+        pmask >>= 1;
+    }
+        
+    // GetProcessAffinityMask can return pmask=0 and smask=0 on systems with more
+    // than 64 processors, which would leave us with a count of 0.  Since the GC
+    // expects there to be at least one processor to run on (and thus at least one
+    // heap), we'll return 64 here if count is 0, since there are likely a ton of
+    // processors available in that case.  The GC also cannot (currently) handle
+    // the case where there are more than 64 processors, so we will return a
+    // maximum of 64 here.
+    if (count == 0 || count > 64)
+        count = 64;
+
+    cCPUs = count;
+            
+    return count;
+}
+
+// Get global memory status
+// Parameters:
+//  ms - pointer to the structure that will be filled in with the memory status
+void GCToOSInterface::GetMemoryStatus(GCMemoryStatus* ms)
+{
+    MEMORYSTATUSEX memStatus;
+    memStatus.dwLength = sizeof(MEMORYSTATUSEX);
+
+    // TODO: fail fast if the function call fails?
+    GlobalMemoryStatusEx(&memStatus);
+
+    // Convert Windows struct to abstract struct
+    ms->dwMemoryLoad     = memStatus.dwMemoryLoad;
+    ms->ullTotalPhys     = memStatus.ullTotalPhys;
+    ms->ullAvailPhys     = memStatus.ullAvailPhys;
+    ms->ullTotalPageFile = memStatus.ullTotalPageFile;
+    ms->ullAvailPageFile = memStatus.ullAvailPageFile;
+    ms->ullTotalVirtual  = memStatus.ullTotalVirtual;
+    ms->ullAvailVirtual  = memStatus.ullAvailVirtual;
+}
+
+// Get a high precision performance counter
+// Return:
+//  The counter value
+int64_t GCToOSInterface::QueryPerformanceCounter()
+{
+    LARGE_INTEGER ts;
+    if (!::QueryPerformanceCounter(&ts))
+    {
+        ASSERT_UNCONDITIONALLY("Fatal Error - cannot query performance counter.");
+        RhFailFast();
+    }
+
+    return ts.QuadPart;
+}
+
+// Get a frequency of the high precision performance counter
+// Return:
+//  The counter frequency
+int64_t GCToOSInterface::QueryPerformanceFrequency()
+{
+    LARGE_INTEGER frequency;
+    if (!::QueryPerformanceFrequency(&frequency))
+    {
+        ASSERT_UNCONDITIONALLY("Fatal Error - cannot query performance counter.");
+        RhFailFast();
+    }
+
+    return frequency.QuadPart;
+}
+
+// Get a time stamp with a low precision
+// Return:
+//  Time stamp in milliseconds
+uint32_t GCToOSInterface::GetLowPrecisionTimeStamp()
+{
+    return ::GetTickCount();
+}
+
+// Parameters of the GC thread stub
+struct GCThreadStubParam
+{
+    GCThreadFunction GCThreadFunction;
+    void* GCThreadParam;
+};
+
+// GC thread stub to convert GC thread function to an OS specific thread function
+static DWORD GCThreadStub(void* param)
+{
+    GCThreadStubParam *stubParam = (GCThreadStubParam*)param;
+    stubParam->GCThreadFunction(stubParam->GCThreadParam);
+
+    return 0;
+}
+
+// Create a new thread
```

Comment:
You should probably callout that this is a thread _for gc use_ and not just any old thread.  You're creating a thread with `THREAD_PRIORITY_HIGHEST`, which not something we want any old thread to use.